pkg/apis/gcp/validation/infrastructure.go

```diff
@@ -84,6 +83,16 @@ func ValidateInfrastructureConfig(infra *apisgcp.InfrastructureConfig, nodesCIDR
 		allErrs = append(allErrs, field.Invalid(networksPath.Child("vpc", "cloudRouter"), infra.Networks.VPC.CloudRouter, "cloud router can not be configured when the VPC name is not specified"))
 	}
 
+	if infra.Networks.VPC != nil && len(infra.Networks.VPC.Name) > 0 {
+		if infra.Networks.VPC.CloudRouter == nil {
```

Comment:
```suggestion
		if infra.Networks.VPC.CloudRouter == nil || len(infra.Networks.VPC.CloudRouter.Name) == 0 {
```

allows you to get rid of lines 91-93

Reply:
you still need another fldpath Error if you want to be specific about the erroneous field 
```go 
networksPath.Child("vpc", "cloudRouter", "name"). 
```